cocotb/share/lib/gpi_log/gpi_logging.cpp

```diff
@@ -125,10 +119,23 @@ extern "C" void gpi_native_logger_vlog(const char *name, int level,
         return;
     }
 
-    int n = vsnprintf(log_buff, LOG_SIZE, msg, argp);
-
-    if (n < 0 || n >= LOG_SIZE) {
-        fprintf(stderr, "Log message construction failed\n");
+    va_list argp_copy;
+    va_copy(argp_copy, argp);
+
+    static std::vector<char> log_buff(512);
+
+    log_buff.clear();
+    size_t n = vsnprintf(log_buff.data(), log_buff.capacity(), msg, argp);
```

Comment:
```suggestion
    int n = vsnprintf(log_buff.data(), log_buff.capacity(), msg, argp);
```

Reply:
Thanks C committee for the great interface... size out should be a size_t and error code should be a separate output.